tICA/scripts/tICAMakeCleanLongitudinalTemplate.sh

```diff
@@ -0,0 +1,158 @@
+opts_AddMandatory '--extract-fmri-name-list' 'concatNamesToUse' 'name@name@name...' "list of fMRI run names to concatenate into the --extract-fmri-out output after tICA cleanup"
+opts_AddMandatory '--highpass' 'HighPass' 'hp value' 'High pass used with these data'
+opts_AddMandatory '--extract-fmri-name' 'extractNameOut' 'name' "fMRI name for concatenated extracted runs.
+Must match the one used in cortical registration."
+opts_AddOptional  '--reg-name' 'RegName' 'registration algorithm' "Cortical registration algorithm used [MSMAll]" "MSMAll"
+opts_AddOptional  '--fmri-name-concat-all' 'extractNameAll' 'name' "Concatenated output run label. If specified, must match the one used in multi-run FIX and cortical registration." ""
+opts_AddOptional  '--fmri-names' 'fMRINames' 'rfMRI_REST1_LR@rfMRI_REST1_RL...' "list of all fmri run names separated by @. Required with --extract-fmri-out-all" ""
+
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+if [[ "$extractNameAll" != "" && "$fMRINames" == "" ]]; then 
+    log_Err_Abort "--fmri-names is required with --fmri-name-concat-all"
+fi
+
+#display the parsed/default values
+opts_ShowValues
+
+#do work
+TemplateDir="$StudyFolder"/"$Subject.long.$TemplateLong"
+
+function makeTemplateConcatRuns {
+    local fMRIStr="$1" sessionsStr="$2" template="$3" nameOut="$4" 
+    local fMRIs sessions
+    IFS=@ read -r -a fMRIs <<< "${fMRIStr}"
+    IFS=@ read -r -a sessions <<< "${sessionsStr}"
+    local numSessions="${#sessions[@]}"
+    local OutDir="$TemplateDir/MNINonLinear/Results/$nameOut"
+    mkdir -p "$OutDir"
+    
+    #per fMRI run lists for concat/merge commands
+    local vn_average_cifti_array
+    local vn_average_nifti_array #average of the _vn from all timepoints
+    local ts_concat_cifti_array
+    local ts_concat_nifti_array #average nifti timeseries
+    
+    #loop variables 
+    local session sessionLong resultsDir fmri fmri_rt frame_ind1 frame_ind2 NumTPs
+    
+    local ts_all_dtseries_clean #concatenated runs from all timepoints in order after demeaning, dividing by _vn and muliplying by the average _vn
+    for session in ${sessions[*]}; do
+        sessionLong="$session.long.$template"
+        echo "Timepoint: $sessionLong"
+        resultsDir="$StudyFolder"/"$sessionLong"/MNINonLinear/Results
+
+        # Average vn's in the atlas and native spaces. We assume that $nameOut exists in $sessionLong tICA output
+        vn_average_cifti_array+=(-cifti "$resultsDir/$nameOut/${nameOut}_Atlas_${RegName}_hp${HighPass}_clean_tclean_vn.dscalar.nii")
+        vn_average_nifti_array+=(-volume "$resultsDir/$nameOut/${nameOut}_hp${HighPass}_clean_tclean_vn.nii.gz")
+        
+        #reset session starting frame
+        frame_ind1=0
+        
+        for fmri in ${fMRIs[*]}; do
```

Comment:
```suggestion
        for fmri in "${fMRIs[@]}"; do
```
If we want to support spaces as additional separators (and therefore can't have spaces in the fmri names), add it to the IFS above like `IFS='@ ' read -r -a ...`.

Reply:
Ok